gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/ExecutionTreeBuilder.java

```diff
@@ -72,6 +86,61 @@ public QueryNode build() {
     return executionTree;
   }
 
+  private QueryNode buildExecutionTreeForSameSourceFilterAndSelection(String source) {
+    if (source.equals("QS")) {
```

Comment:
why is this source string hardcoded? Don't we have some enum or some constant somewhere?

Reply:
Will use AttributeSource#name()